### PR TITLE
signup check username and verifiedemail

### PR DESCRIPTION
**Pull request by TdevM**

### CODE REVIEW

**Comment by championswimmer** [comment]
This pull request **introduces 1 alert** when merging feef56a29c667d0967926b8c802b2ce1081f2800 into 4be0b0359776dfe2085c59d79c6fa29c42109096 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-886ddffc7651dfc47eda579c883e9ddbba6926c1)

**new alerts:**

* 1 for Type confusion through parameter tampering

---

**Warning** - Automated code review for coding-blocks/oneauth will be disabled on October 1, 2019. You can avoid this by <a href="https://github.com/apps/lgtm-com/installations/new">installing the LGTM.com GitHub App</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by championswimmer** [comment]
This pull request **introduces 1 alert** when merging d015625c442b8a055354a104b326787cd7c6c49a into 4be0b0359776dfe2085c59d79c6fa29c42109096 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-357fc9b1be4cb60a8e0aa89b5835b0a127019642)

**new alerts:**

* 1 for Type confusion through parameter tampering

---

**Warning** - Automated code review for coding-blocks/oneauth will be disabled on October 1, 2019. You can avoid this by <a href="https://github.com/apps/lgtm-com/installations/new">installing the LGTM.com GitHub App</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/routers/api/signupcheck.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        try {
+            const user = await findUserByParams({username: req.query.username});
+            if (user || req.query.username.length < 3 || validateUsername(req.query.username)) {
+                res.status(422).json()
+            } else {
+                res.status(200).json()
+            }
+        } catch (error) {
+            Raven.captureException(error)
+            res.status(400).json({error: 'Failed to search username'})
```
Comment (witty123) [inline src/routers/api/signupcheck.js]: 500

File: src/routers/api/signupcheck.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        try {
+            const user = await findUserByParams({verifiedemail: req.query.email});
+            if (user || !emailRegex.test(req.query.email)) {
+                res.status(422).json()
+            } else {
+                res.status(200).json()
+            }
+        } catch (error) {
+            Raven.captureException(error)
+            res.status(400).json({error: 'Failed to search email'})
```
Comment (witty123) [inline src/routers/api/signupcheck.js]: 500

File: src/routers/api/signupcheck.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    })
+
+router.get('/email',
+    passport.authenticate('bearer', {session: false}),
+    async function (req, res, next) {
+        let emailRegex = /^[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$/i;
+        try {
+            const user = await findUserByParams({verifiedemail: req.query.email});
+            if (user || !emailRegex.test(req.query.email)) {
+                res.status(422).json()
```
Comment (witty123) [inline src/routers/api/signupcheck.js]: Add a message saying email already exists, always good to be descriptive with messages.

File: src/routers/api/signupcheck.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+const {findUserByParams} = require('../../controllers/user')
+const {validateUsername}  = require('../../utils/username_validator')
+
+router.get('/username',
+    passport.authenticate('bearer', {session: false}),
+    async function (req, res, next) {
+        try {
+            const user = await findUserByParams({username: req.query.username});
+            if (user || req.query.username.length < 3 || validateUsername(req.query.username)) {
+                res.status(422).json()
```
Comment (witty123) [inline src/routers/api/signupcheck.js]: Add a message saying username already exists, always good to be descriptive with messages.

File: src/routers/api/signupcheck.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            } else {
+                res.status(200).json()
```
Comment (witty123) [inline src/routers/api/signupcheck.js]: Similar message

**Comment by championswimmer** [comment]
This pull request **introduces 1 alert** when merging 92b9235544337059db4ac96d036c6c3127372426 into 4be0b0359776dfe2085c59d79c6fa29c42109096 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-a4d8fadbabbc2b7133f1959d76999e669372ae82)

**new alerts:**

* 1 for Type confusion through parameter tampering

---

**Warning** - Automated code review for coding-blocks/oneauth will be disabled on October 1, 2019. You can avoid this by <a href="https://github.com/apps/lgtm-com/installations/new">installing the LGTM.com GitHub App</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by championswimmer** [comment]
This pull request **introduces 1 alert** when merging c0ce20070d4e6a990607e81ba7de23ff5cac252d into 260d43a32f061c74afee7503694aae7bc4c2d872 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-fcf601c72a289998235fb7f1d02c5912c9cbd520)

**new alerts:**

* 1 for Type confusion through parameter tampering

---

**Warning** - Automated code review for coding-blocks/oneauth will be disabled on October 1, 2019. You can avoid this by <a href="https://github.com/apps/lgtm-com/installations/new">installing the LGTM.com GitHub App</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by championswimmer** [comment]
This pull request **introduces 1 alert** when merging 3bc234a278565dfd28b4da3b06c1dcdb77b690b6 into 260d43a32f061c74afee7503694aae7bc4c2d872 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-b524c2a23101f2eb86348ecac3374e9dff458a9d)

**new alerts:**

* 1 for Type confusion through parameter tampering

---

**Warning** - Automated code review for coding-blocks/oneauth will be disabled on October 1, 2019. You can avoid this by <a href="https://github.com/apps/lgtm-com/installations/new">installing the LGTM.com GitHub App</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**